### PR TITLE
Switch to SUSE Enterprise Storage 2.1

### PR DESCRIPTION
**Pull request by mjura**

### CODE REVIEW

File: crowbar_framework/app/models/ceph_service.rb [hunk 1]
```diff
@@ -49,7 +49,8 @@ def role_constraints
           "unique" => false,
           "count" => 1,
           "platform" => {
-            "suse" => "12.0"
+            "suse" => "12.0",
+            "suse" => "12.1"
```
Comment (tboerger) [inline crowbar_framework/app/models/ceph_service.rb:52]: As HoundCI say this won't work. You can not have multiple same keys :)

Comment (tboerger) [inline crowbar_framework/app/models/ceph_service.rb:52]: If you want to support multiple versions you are forced to use regular expressions like `/12.[01]{1}/`

File: crowbar_framework/app/models/ceph_service.rb [hunk 2]
```diff
@@ -57,21 +58,24 @@ def role_constraints
           "unique" => false,
           "count" => 3,
           "platform" => {
-            "suse" => "12.0"
+            "suse" => "12.0",
+            "suse" => "12.1"
```
Comment (tboerger) [inline crowbar_framework/app/models/ceph_service.rb:60]: Same as above

File: crowbar_framework/app/models/ceph_service.rb [hunk 3]
```diff
@@ -57,21 +58,24 @@ def role_constraints
...
           },
         },
         "ceph-osd" => {
           "unique" => false,
           "count" => 8,
           "platform" => {
-            "suse" => "12.0"
+            "suse" => "12.0",
+            "suse" => "12.1"
```
Comment (tboerger) [inline crowbar_framework/app/models/ceph_service.rb:67]: Same as above

File: crowbar_framework/app/models/ceph_service.rb [hunk 4]
```diff
@@ -57,21 +58,24 @@ def role_constraints
...
           },
         },
         "ceph-radosgw" => {
           "unique" => false,
           "count" => 1,
           "platform" => {
-            "suse" => "12.0"
+            "suse" => "12.0",
+            "suse" => "12.1"
```
Comment (tboerger) [inline crowbar_framework/app/models/ceph_service.rb:74]: Same as above

File: crowbar_framework/app/models/ceph_service.rb [hunk 5]
```diff
@@ -49,7 +49,7 @@ def role_constraints
           "unique" => false,
           "count" => 1,
           "platform" => {
-            "suse" => "12.0"
+            "suse" => "< 12.1"
```
Comment (dirkmueller) [inline crowbar_framework/app/models/ceph_service.rb]: that should be "12.1" I think..

File: crowbar_framework/app/models/ceph_service.rb [hunk 6]
```diff
@@ -57,21 +57,21 @@ def role_constraints
           "unique" => false,
           "count" => 3,
           "platform" => {
-            "suse" => "12.0"
+            "suse" => "< 12.1"
```
Comment (dirkmueller) [inline crowbar_framework/app/models/ceph_service.rb]: that should be "12.1" I think..

File: crowbar_framework/app/models/ceph_service.rb [hunk 7]
```diff
@@ -57,21 +57,21 @@ def role_constraints
...
           },
         },
         "ceph-osd" => {
           "unique" => false,
           "count" => 8,
           "platform" => {
-            "suse" => "12.0"
+            "suse" => "< 12.1"
```
Comment (dirkmueller) [inline crowbar_framework/app/models/ceph_service.rb]: that should be "12.1" I think..

File: crowbar_framework/app/models/ceph_service.rb [hunk 8]
```diff
@@ -57,21 +57,21 @@ def role_constraints
...
           },
         },
         "ceph-radosgw" => {
           "unique" => false,
           "count" => 1,
           "platform" => {
-            "suse" => "12.0"
+            "suse" => "< 12.1"
```
Comment (dirkmueller) [inline crowbar_framework/app/models/ceph_service.rb]: that should be "12.1" I think..

**Comment by mjura** [comment]
@dirkmueller : Please correct me, but I think that should be better if we allow to install SES 2.0 and SES 2.1. I've change regexp. I'm fine also with fixed version 12.1

File: crowbar_framework/app/models/ceph_service.rb [hunk 9]
```diff
@@ -49,7 +49,7 @@ def role_constraints
           "unique" => false,
           "count" => 1,
           "platform" => {
-            "suse" => "12.0"
+            "suse" => "/^12.[01]{1}$/"
```
Comment (dirkmueller) [inline crowbar_framework/app/models/ceph_service.rb]: 12.1

File: crowbar_framework/app/models/ceph_service.rb [hunk 10]
```diff
@@ -57,21 +57,21 @@ def role_constraints
           "unique" => false,
           "count" => 3,
           "platform" => {
-            "suse" => "12.0"
+            "suse" => "/^12.[01]{1}$/"
```
Comment (dirkmueller) [inline crowbar_framework/app/models/ceph_service.rb]: 12.1

File: crowbar_framework/app/models/ceph_service.rb [hunk 11]
```diff
@@ -57,21 +57,21 @@ def role_constraints
...
           },
         },
         "ceph-osd" => {
           "unique" => false,
           "count" => 8,
           "platform" => {
-            "suse" => "12.0"
+            "suse" => "/^12.[01]{1}$/"
```
Comment (dirkmueller) [inline crowbar_framework/app/models/ceph_service.rb]: 12.1

File: crowbar_framework/app/models/ceph_service.rb [hunk 12]
```diff
@@ -57,21 +57,21 @@ def role_constraints
...
           },
         },
         "ceph-radosgw" => {
           "unique" => false,
           "count" => 1,
           "platform" => {
-            "suse" => "12.0"
+            "suse" => "/^12.[01]{1}$/"
```
Comment (dirkmueller) [inline crowbar_framework/app/models/ceph_service.rb]: 12.1

**Comment by dirkmueller** [comment]
@mjura we'll only test with and be compatible with SES 2.1.

**Comment by mjura** [comment]
Ok, thanks

**Comment by toabctl** [comment]
+1

**Comment by aplanas** [comment]
Gating fails with when `zypper-retry --non-interactive install --auto-agree-with-licenses ceph` with `ceph' not found in package names. Trying capabilities.`

**Comment by vuntz** [comment]
+1

**Comment by mjura** [comment]
Gating failing because this PR depends on https://github.com/crowbar/crowbar-core/pull/130 and https://github.com/SUSE-Cloud/automation/pull/656

**Pull request merged**